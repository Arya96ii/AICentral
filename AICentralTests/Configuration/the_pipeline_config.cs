using System.Text;
using AICentral;
using AICentral.Configuration;
using AICentral.Core;
using AICentral.Logging.AzureMonitor;
using AICentral.OpenAI.AzureOpenAI;
using ApprovalTests;
using Microsoft.AspNetCore.Builder;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Newtonsoft.Json;

namespace AICentralTests;

public class the_pipeline_config
{
    [Fact]
    public void supports_minimal_setup()
    {
        using var stream = new MemoryStream(
            Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(new
            {
                AICentral = new
                {
                    Endpoints = new[]
                    {
                        new
                        {
                            Type = "AzureOpenAIEndpoint",
                            Name = "test-endpoint",
                            Properties = new AzureOpenAIEndpointPropertiesConfig()
                            {
                                ApiKey = "1234",
                                AuthenticationType = "ApiKey",
                                LanguageEndpoint = "https://somehere.com",
                                ModelMappings = new Dictionary<string, string>()
                            }
                        }
                    },
                    EndpointSelectors = new[]
                    {
                        new
                        {
                            Type = "SingleEndpoint",
                            Name = "default-endpoint-selector",
                            Properties = new
                            {
                                Endpoint = "test-endpoint"
                            }
                        }
                    },
                    AuthProviders = new[]
                    {
                        new
                        {
                            Type = "AllowAnonymous",
                            Name = "anonymous"
                        }
                    },
                    Pipelines = new[]
                    {
                        new AICentralPipelineConfig()
                        {
                            Name = "test-pipeline",
                            Host = "my-test-host.localtest.me",
                            AuthProvider = "anonymous",
                            EndpointSelector = "default-endpoint-selector",
                        }
                    },
                }
            }))
        );

        var host = WebApplication.CreateBuilder(new WebApplicationOptions() {EnvironmentName = "test"});
        host.Configuration.AddJsonStream(stream);
        host.Services.AddAICentral(host.Configuration,
            additionalComponentAssemblies: typeof(AzureMonitorLogger).Assembly);
        var app = host.Build();

        var pipelines = app.Services.GetRequiredService<ConfiguredPipelines>();
        Approvals.VerifyJson(JsonConvert.SerializeObject(pipelines.WriteDebug(), Formatting.Indented));
    }
}