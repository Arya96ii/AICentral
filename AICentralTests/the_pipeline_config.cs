using System.Text;
using AICentral;
using AICentral.Configuration;
using AICentral.Configuration.JSON;
using ApprovalTests;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Newtonsoft.Json;

namespace AICentralTests;

public class the_pipeline_config
{
    [Fact]
    public void supports_minimal_setup()
    {
        using var stream = new MemoryStream(
            Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(new
            {
                AICentral = new
                {
                    Endpoints = new[]
                    {
                        new
                        {
                            Type = "AzureOpenAIEndpoint",
                            Name = "test-endpoint",
                            Properties = new ConfigurationTypes.AICentralPipelineAzureOpenAIEndpointPropertiesConfig()
                            {
                                ApiKey = "1234",
                                LanguageEndpoint = "https://somehere.com",
                                ModelMappings = new Dictionary<string, string>()
                            }
                        }
                    },
                    EndpointSelectors = new[]
                    {
                        new
                        {
                            Type = "SingleEndpoint",
                            Name = "default-endpoint-selector",
                            Properties = new
                            {
                                Endpoint = "test-endpoint"
                            }
                        }
                    },
                    Pipelines = new[]
                    {
                        new ConfigurationTypes.AICentralPipelineConfig()
                        {
                            Name = "test-pipeline",
                            Host = "my-test-host.localtest.me",
                            EndpointSelector = "default-endpoint-selector",
                        }
                    }
                }
            }))
        );

        var host = WebApplication.CreateBuilder();
        host.Configuration.AddJsonStream(stream);
        host.Services.AddAICentral(host.Configuration);
        var app = host.Build();

        var pipelines = app.Services.GetRequiredService<AICentralPipelines>();
        Approvals.VerifyJson(JsonConvert.SerializeObject(pipelines.WriteDebug(), Formatting.Indented));
    }
}