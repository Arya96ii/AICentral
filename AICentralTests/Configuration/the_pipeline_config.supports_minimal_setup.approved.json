[
    
  {
        
    "Name":  "test-pipeline",
        
    "RouteMatch":  {
            
      "Host":  "my-test-host.localtest.me"
    
        },
        
    "ClientAuth":  {
            
      "auth":  "No Consumer Auth"
    
        },
        
    "Steps":  [],
        
    "EndpointSelector":  {
            
      "Type":  "SingleEndpoint",
            
      "Endpoints":  [
                
        {
                    
          "Type":  "AzureOpenAI",
                    
          "Url":  "https://somehere.com",
                    
          "Mappings":  {},
                    
          "Auth":  {
                        
            "Type":  "ApiKey"
          
                    }
        
                }
      
            ]
    
        }
  
    }

]